engine/Default/forces_attack_processing.php

```diff
@@ -41,35 +58,55 @@
 
 $results = array('Attackers' => array('TotalDamage' => 0),
 				'Forces' => array(),
-				'Forced' => false);
+				'Forced' => $bump);
 
 $sector =& $player->getSector();
-$attackers =& $sector->getFightingTradersAgainstForces($player, $forces);
+if ($bump) {
+	//When hitting mines by bumping only the current player attacks/gets hit.
+	$attackers = array(&$player);
+} else {
+	$attackers =& $sector->getFightingTradersAgainstForces($player, $forces);
+}
 
 //decloak all attackers
 foreach($attackers as &$attacker) {
 	$attacker->getShip()->decloak();
-	$attacker->setLastSectorID(0);
+	if (!$bump) {
+		$attacker->setLastSectorID(0);
+	}
 } unset($attacker);
 
+// If mines are bumped, the forces shoot first. Otherwise player shoots first.
+if ($bump) {
+	$results['Forces'] =& $forces->shootPlayers($attackers,true);
```

Comment:
The second param of both of these `$forces->shootPlayers` can be `$bump`, as that's what it's intended to denote anyway

Reply:
Done!